components/server/src/database/measurements.py

```diff
@@ -17,20 +18,43 @@ def latest_measurement(database: Database, metric: Metric) -> Measurement | None
     return None if latest is None else Measurement(metric, latest)
 
 
+def latest_measurements_by_metric_uuid(database: Database, metric_uuids: List[str]) -> Dict[str, Measurement] | None:
+    """Return the latest measurements in a dict with metric_uuids as keys."""
+    latest_measurement_ids = database.measurements.aggregate(
+        [
+            {"$match": {"metric_uuid": {"$in": metric_uuids}}},
+            {"$sort": {"metric_uuid": 1, "start": -1}},
+            {"$group": {"_id": "$metric_uuid", "measurement_id": {"$first": "$_id"}}},
```

Comment:
Is an alternative to copy all measurement fields (instead of just the measurement_id) as part of the `$group` step? Something like:
```suggestion
            {"$group": {"_id": "$metric_uuid", "metric_uuid": {"$first": "$metric_uuid"}, "sources": {"$first": "$sources"}, "start": {"$first": "$start"}, "end": {"$first": "$end"}, etc...},
```

That way the separate find isn't necessary and the result of the aggregation would basically be a list of measurements. A disadvantage of this approach is that the measurement fields need to be listed explicitly.